### PR TITLE
Decode Import Table Sections

### PR DESCRIPTION
**Pull request by commonlisp**
The new import table entries contain both an imported module name and function name as well as a signature index.

### CODE REVIEW

File: lib/WasmReader/WasmBinaryReader.cpp [hunk 1]
```diff
@@ -652,6 +662,28 @@ WasmBinaryReader::FunctionBodyHeader()
...
+WasmBinaryReader::ImportEntry()
+{
+    UINT len = 0;
+    UINT sigId = LEB128(len);
+
+    m_funcInfo = Anew(&m_alloc, WasmFunctionInfo, &m_alloc);
+    m_funcInfo->SetModuleName(ReadString());
+    m_funcInfo->SetName(ReadString());
+
+    m_currentNode.func.info = m_funcInfo;
```
Comment (MikeHolman) [inline lib/WasmReader/WasmBinaryReader.cpp]: this is not used... we should create another table of WasmFunctionInfo for imports on ModuleInfo.

File: lib/WasmReader/WasmByteCodeGenerator.cpp [hunk 2]
```diff
@@ -87,7 +91,7 @@ WasmBytecodeGenerator::GenerateModule()
     {
         if (m_reader->ReadNextSection((SectionCode)sectionCode))
         {
-            if (!sectionProcess[sectionCode](this, (SectionCode)sectionCode))
+            if (sectionProcess[sectionCode](this, (SectionCode)sectionCode) == psrInvalid)
```
Comment (Cellule) [inline lib/WasmReader/WasmByteCodeGenerator.cpp]: This should actually be `!= psrEnd`

**Comment by Cellule** [comment]
I think we should remove the `import` concept from the WasmFunctionInfo since they are no longer associated.
the operator `call_import` uses an index of the import table and not the function table, therefor they need to be stored separately.
I think we should do something similar to what I did for exports (see #489)
We'll have to allocate some space on the frame for the imports.

**Comment by commonlisp** [comment]
Ok, I'll see what I can do to follow that closely.

File: lib/WasmReader/WasmByteCodeGenerator.cpp [hunk 3]
```diff
@@ -496,7 +500,7 @@ WasmBytecodeGenerator::EmitCall()
 
     uint funcNum = Js::Constants::UninitializedValue;
     uint signatureId = Js::Constants::UninitializedValue;
-    bool imported;
+    bool imported = false;
```
Comment (MikeHolman) [inline lib/WasmReader/WasmByteCodeGenerator.cpp]: can remove this completely

File: lib/Runtime/Base/ScriptContext.cpp [hunk 4]
```diff
@@ -1952,6 +1919,34 @@ namespace Js
...
+                    Js::Throw::InternalError();
+                }
+                Var modProp = JavascriptOperators::OP_GetProperty(ffi, modPropertyRecord->GetPropertyId(), this);
+                if (!JavascriptFunction::Is(modProp))
+                {
+                    Assert(UNREACHED);
+                    // TODO: michhol figure out correct error path
+                }
+                Var prop = JavascriptOperators::OP_GetProperty(modProp, propertyRecord->GetPropertyId(), this);
+                localModuleFunctions[i] = prop; // TODO: Fix this.
```
Comment (MikeHolman) [inline lib/Runtime/Base/ScriptContext.cpp]: fix this?

**Comment by MikeHolman** [comment]
lgtm

**Pull request merged**